apps/api/src/coscrad-cli/ingest-media-items.cli-command.ts

```diff
@@ -67,10 +73,36 @@ export class IngestMediaItemsCliCommand extends CliCommandRunner {
 
         const generatedIds = await this.idManager.generateMany(partialPayloads.length);
 
+        const mediaLengthMap = new Map<string, number>();
+
+        for (const { filename, mimeType } of partialPayloads) {
+            const ffprobeResult = await ffprobe(`${directory}/${filename}`);
+
+            const { streams } = ffprobeResult;
+
+            const duration = streams?.[0].duration;
+
+            if (
+                !isNullOrUndefined(duration) &&
+                (isVideoMimeType(mimeType) || isAudioMimeType(mimeType))
+            ) {
+                mediaLengthMap.set(filename, duration);
+            }
+        }
+
         const createMediaItemFsas: CommandFSA<CreateMediaItem>[] = partialPayloads.map(
             (partialFsa, index) => {
                 const { filename } = partialFsa;
 
+                const lengthSeconds = mediaLengthMap.get(filename);
+
+                // TODO Use a math lib
```

Comment:
Add a story for this.

Reply:
[Here](https://github.com/COSCRAD/coscrad/pull/520#discussion_r1431795341) is the story.